crowdfunding/users/serializers.py

```diff
@@ -0,0 +1,38 @@
+from rest_framework import serializers
+
+import random
+import string
+
+from .models import CustomUser
+
+def random_username():
+    letters = string.ascii_lowercase
+    result_username = ''.join(random.choice(letters) for i in range(7))
+    return result_username
+
+class CustomUserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CustomUser
+        fields = ['email', 'username', 'password']
+        read_only_fields = ['username']
+        # extra_kwargs = {'password': {'write_only': True}}
+
+    def create(self, validated_data):
+        user = CustomUser(
+            email=validated_data['email'],
+            username=random_username(),
+        )
+        user.set_password(validated_data['password'])
+        user.save()
+        return user
+
+    def update(self, instance, validated_data):
+        instance.email = validated_data.get('email', instance.email)
+        instance.set_password(validated_data['password'])
+        instance.save()
+        return instance
+
+class ChangePasswordSerializer(serializers.Serializer):
+    model = CustomUser
```

Comment:
you shouldn't need to supply a model for a regular serializer, only for a model serializer